src/lib/components/Tabs/index.js

```diff
@@ -94,8 +96,119 @@ class Tabs extends BaseElement {
     `;
   }
 
+  panelTemplate(i, child) {
+    return html`
+      <div
+        id="web-tab-${this.idSalt}-${i}-panel"
+        class="web-tabs__panel"
+        role="tabpanel"
+        aria-labelledby="web-tab-${this.idSalt}-${i}"
+        hidden
+      >
+        ${child}
+      </div>
+    `;
+  }
+
+  firstUpdated() {
+    this.activeTab = 0;
+    this.classList.remove("unresolved");
+    this.onResize();
+
+    // If Tabs component contains AssessmentQuestion components,
+    // listen for requests to navigate to the next tab.
+    const questions = this.querySelectorAll("web-question");
+
+    if (!questions) return;
+    for (const question of questions) {
+      question.addEventListener("request-nav-to-next", this.nextTab);
+    }
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener("resize", this.onResize);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    window.removeEventListener("resize", this.onResize);
+  }
+
+  set activeTab(val) {
+    const oldVal = this.activeTab_;
+
+    this.activeTab_ = Math.floor(val);
+
+    this.changeTab();
+    this.requestUpdate("activeTab", oldVal);
+  }
+
+  get activeTab() {
+    return this.activeTab_;
+  }
+
+  // Update state of tabs and associated panels.
+  changeTab() {
+    const tabs = this.querySelectorAll(".web-tabs__tab");
+    const panels = this.querySelectorAll(".web-tabs__panel");
+    const activeTab = tabs[this.activeTab];
+
+    if (!panels[this.activeTab]) return;
```

Comment:
That's fair

Reply:
Well, actually, I'm trying to think this through: if you somehow end up in a state where there's a mismatch between tabs and panels, will it be worse to change the tab state when there's no corresponding panel? It might, for example, lead you to believe that two tabs have the same content. If the tab state doesn't change, that might be more of a hint that something's wrong?

Then again, I'm not really sure how you'd end up in this state unless you're monkeying with the DevTools Elements panel... Open to suggestions about how to handle this error!

Reply:
I think you should just change the code below to be a bit more permissive.

wrap the `activeTab` changes:

```js
if (activeTab) {
  activeTab.setAttribute(...)
}
```

and

```
const activePanel = panels[this.activeTab];
if (activePanel) {
  activePanel.hidden = false;
}
```

Then you can drop this if.

I know it's not supposed to happen, but I think this is more sane.